FWCore/Utilities/src/compactStringSerializer.cc

```diff
@@ -0,0 +1,11 @@
+#include "FWCore/Utilities/interface/compactStringSerializer.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+namespace edm::compactString::detail {
+  void throwIfContainsDelimiters(std::string const& str) {
+    auto pos = str.find_first_of(kDelimiters);
+    if (pos != std::string::npos) {
+      throw cms::Exception("compactString") << "Serialized string '" << str << "' contains a delimiter";
```

Comment:
How about saying what the position of the delimiter is and which one? The problem is printing it will not show it as we are using non-printable characters.

Reply:
Good idea